doc/conf.py

```diff
@@ -122,6 +122,17 @@
 # A list of ignored prefixes for module index sorting.
 #modindex_common_prefix = []
 
+## Substitutions to be used in all documents:
+
+# loading variables
+conftest_vars = {}
+with open(os.path.join('..', 'conftest.py')) as fp:
+    exec(fp.read(), conftest_vars)
```

Comment:
Not so keen on the exec, and avoiding the exec would probably lead to some possibly awkward imports / PATH issues.

Having this mechanism to avoid putting the version once in a rst is a bit too much IMO :) If the version changes we would have to change the readme anyway, so we might as well do it twice (which would need to be done twice in any case) . `git grep 3.3.0` doesn't cost much..

Reply:
Better off having a `sklearn/_requirements.py` file than this... but this is all too much sophistication for the present PR. Thanks for trying to makde it maintainable!